engine/runtime-compiler/src/main/java/org/enso/compiler/dump/DocsEmitMarkdown.java

```diff
@@ -0,0 +1,59 @@
+package org.enso.compiler.dump;
+
+import java.io.IOException;
+import org.enso.compiler.core.IR;
+import org.enso.compiler.core.ir.Module;
+import org.enso.compiler.core.ir.module.scope.Definition;
+import org.enso.compiler.core.ir.module.scope.definition.Method;
+import org.enso.compiler.pass.resolve.DocumentationComments;
+import org.enso.compiler.pass.resolve.DocumentationComments$;
+import org.enso.pkg.QualifiedName;
+
+/** Visitor that emits documentation in markdown format. */
+final class DocsEmitMarkdown implements DocsVisit {
+
+  @Override
+  public boolean visitUnknown(IR ir, Appendable w) {
+    return true;
+  }
+
+  @Override
+  public boolean visitModule(QualifiedName name, Module module, Appendable w) throws IOException {
+    w.append("## Documentation for " + name + "\n");
+    writeDocs(module, w);
+    return true;
+  }
+
+  @Override
+  public void visitMethod(Definition.Type t, Method.Explicit m, Appendable w) throws IOException {
+    w.append("#### method " + m.methodName().name() + "\n");
+    writeDocs(m, w);
+  }
+
+  @Override
+  public void visitConversion(Method.Conversion c, Appendable w) throws IOException {
+    w.append("#### conversion " + c.methodName().name() + "\n");
+    writeDocs(c, w);
+  }
+
+  private void writeDocs(IR b, Appendable w) throws IOException {
+    var option = b.passData().get(DocumentationComments$.MODULE$);
+    if (option.isDefined()) {
+      var doc = (DocumentationComments.Doc) option.get();
+      w.append(doc.documentation());
+      w.append("\n\n\n");
```

Comment:
OS-specific new line?

Reply:
Are you suggesting we should use platform specific end of lines rather than `\n` all the time?
- then we should use either [BufferedWriter.newLine()](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/io/BufferedWriter.html#newLine())
- or [PrintWriter.println](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/io/PrintWriter.html#println(java.lang.String))
- instead of including `\n` in the generated texts

If so, then there is much more code to change than this single line...

Reply:
Jaroslav is right. I have encountered this issue before as well. We are hardcoding `\n` all over the place. It seems that the general rule of a thumb is that we are trying to read and write everything with `\n` instead of `\r\n`. On the other hand, using `System.lineSeparator()` instead of hardcoding `\n` is straightforward and should help in the future.

Reply:
I removed all `\n` from this PR in 5c02828